testing/scenario_app/compile_ios_jit.sh

```diff
@@ -90,9 +90,17 @@ echo "Using $SYSROOT as sysroot."
 
 echo "Creating stub App using $SYSROOT..."
 
+# Use the ld specified in PATH instead of the one from the active xcode
+# toolchain (the default).
+#
+# When linking against a newer iOS SDK (newer SYSROOT) while using an older
+# version of clang, the linker from the older toolchain may refuse to link
+# against new TAPI files provided in the new SDK.
+# See: https://github.com/flutter/flutter/issues/65901
 echo "static const int Moo = 88;" | xcrun clang -x c \
   -arch x86_64 \
   -fembed-bitcode-marker \
+  -fuse-ld=$(which ld) \
```

Comment:
I think this will probably cause problems at some point - like if we roll the clang toolchain and end up with an imcompatible version of LD for some reaon, or if we update something about the environment of the buildbots and the path var changes. 

Perhaps instead of using `xcrun` here, we should just use `clang` from our buildroot, which should then use the ld shipped with the buildroot.  WDYT?

Reply:
/cc @jmagman for input on this

Reply:
Tried using the clang located in `src/buildtools/mac-x64/`, it worked with the `PATH` ld trick, its `lld` seems to be a shim. That sounds more future-proof.

Just curious, which clang does the GN build use if `use_xcode` is set? The one in `PATH` right? Not sure why it works if that's the case.

Reply:
Rather than using LD from path, why not just use LD from the buildroot?

if use_xcode is set it should be using whatever `xcrun --find clang` says.

Reply:
There's no `ld` in `src/buildtools/mac-x64/clang/bin`, the `lld` binary in that folder seems to be just a shim.  So to override the `ld` to use for linking the scenario app dylib we still need to add a path prefix.